circleci.sh

```diff
@@ -89,7 +89,7 @@ coverage() {
 
     # rebuild dmd with coverage enabled
     # use the just build dmd as host compiler this time
-    mv src/dmd src/host_dmd
+    mv generated/dmd src/host_dmd
```

Comment:
> So, if this PR goes further all the scripts in the dlang repo must be updated. Is this hard to accomplish?

It's quite a lot of script. And that's only for the visible part, some people might rely on it without you knowing.
Also this change is not a one time cost: for example Digger has to know where the binary lives, so if it's a hard switch it will have to be hard-coded somehow.
I would recommend to have both running side-by-side: First generate all binaries / temp files in a separate folder, and copy over `generated/dmd` to `src/dmd`. Adjust other places, and let a bit of time pass before changing dmd source folder.

I know it's a pain for you, but it will ease the transition for everyone.